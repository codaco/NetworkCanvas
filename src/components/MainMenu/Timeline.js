import React from 'react';
import PropTypes from 'prop-types';
import Scroller from '../../components/Scroller';
import TimelineStage from '../../containers/MainMenu/TimelineStage';

const Timeline = ({ items }) => {
  const menuItems = items.map((item, index) =>
    (<TimelineStage item={item} key={item.id} index={index} />),
  );

  return (
    <div className="stages-timeline__timeline">
      <Scroller>
        { menuItems }
      </Scroller>
    </div>
  );
};

Timeline.propTypes = {
  items: PropTypes.array.isRequired,
};

export default Timeline;
