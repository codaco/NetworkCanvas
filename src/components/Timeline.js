import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import PropTypes from 'prop-types';
import { Icon } from '../ui/components';
import { ProgressBar } from '../components';
import { DropObstacle } from '../behaviours/DragAndDrop';
import { actionCreators as uiActions } from '../ducks/modules/ui';

class Timeline extends PureComponent {
  render() {
    const {
      onClickBack,
      onClickNext,
      percentProgress,
      toggleMenu,
    } = this.props;

    return (
      <div className="timeline" onClick={toggleMenu}>
        <div className="timeline-nav timeline-nav--back">
          <Icon
            onClick={(e) => {
              if (e) {
                e.stopPropagation();
                e.preventDefault();
              }
              onClickBack(e);
            }}
            name="chevron-up"
          />
        </div>
        <ProgressBar percentProgress={percentProgress} />
        <div
          className="timeline-nav timeline-nav--next"
          onClick={(e) => {
            if (e) {
              e.stopPropagation();
              e.preventDefault();
            }
            onClickNext(e);
          }}
        >
          <Icon name="chevron-down" />
        </div>
      </div>
    );
  }
}

Timeline.propTypes = {
  onClickBack: PropTypes.func,
  onClickNext: PropTypes.func,
  percentProgress: PropTypes.number,
  toggleMenu: PropTypes.func.isRequired,
};

Timeline.defaultProps = {
  onClickBack: () => {},
  onClickNext: () => {},
  percentProgress: 0,
};

const mapDispatchToProps = dispatch => ({
  toggleMenu: () => dispatch(uiActions.toggle('isMenuOpen')),
});

export { Timeline };

export default compose(
  connect(null, mapDispatchToProps),
  DropObstacle,
)(Timeline);
