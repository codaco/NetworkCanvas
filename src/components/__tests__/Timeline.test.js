/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Timeline } from '../Timeline';

describe('Timeline component', () => {
  const toggleMock = jest.fn();
  const backMock = jest.fn();
  const nextMock = jest.fn();

  const component = shallow(
    <Timeline
      percentProgress="40"
      onClickBack={backMock}
      onClickNext={nextMock}
    />,
  );

  it('renders Timeline', () => {
    expect(component).toMatchSnapshot();
  });

  it('toggles menu on timeline click', () => {
    expect(toggleMock.mock.calls.length).toBe(0);
    component.find('.timeline').simulate('click');
    expect(toggleMock.mock.calls.length).toBe(1);
  });

  it('calls back function on clicking back button', () => {
    expect(backMock.mock.calls.length).toBe(0);
    component.find('Icon').at(0).simulate('click');
    expect(backMock.mock.calls.length).toBe(1);
  });

  it('calls next function on clicking next button', () => {
    expect(nextMock.mock.calls.length).toBe(0);
    component.find('.timeline-nav--next').simulate('click');
    expect(nextMock.mock.calls.length).toBe(1);
  });
});
