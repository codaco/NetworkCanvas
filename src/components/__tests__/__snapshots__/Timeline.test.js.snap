// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline component renders Timeline 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeline
    onClickBack={[MockFunction]}
    onClickNext={[MockFunction]}
    percentProgress="40"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="timeline-nav timeline-nav--back"
        >
          <Icon
            className=""
            color=""
            name="chevron-up"
            onClick={[Function]}
            style={Object {}}
          />
        </div>,
        <ProgressBar
          percentProgress="40"
        />,
        <div
          className="timeline-nav timeline-nav--next"
          onClick={[Function]}
        >
          <Icon
            className=""
            color=""
            name="chevron-down"
            style={Object {}}
          />
        </div>,
      ],
      "className": "timeline",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            className=""
            color=""
            name="chevron-up"
            onClick={[Function]}
            style={Object {}}
          />,
          "className": "timeline-nav timeline-nav--back",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "color": "",
            "name": "chevron-up",
            "onClick": [Function],
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "percentProgress": "40",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            className=""
            color=""
            name="chevron-down"
            style={Object {}}
          />,
          "className": "timeline-nav timeline-nav--next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "color": "",
            "name": "chevron-down",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="timeline-nav timeline-nav--back"
          >
            <Icon
              className=""
              color=""
              name="chevron-up"
              onClick={[Function]}
              style={Object {}}
            />
          </div>,
          <ProgressBar
            percentProgress="40"
          />,
          <div
            className="timeline-nav timeline-nav--next"
            onClick={[Function]}
          >
            <Icon
              className=""
              color=""
              name="chevron-down"
              style={Object {}}
            />
          </div>,
        ],
        "className": "timeline",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              className=""
              color=""
              name="chevron-up"
              onClick={[Function]}
              style={Object {}}
            />,
            "className": "timeline-nav timeline-nav--back",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "color": "",
              "name": "chevron-up",
              "onClick": [Function],
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "percentProgress": "40",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              className=""
              color=""
              name="chevron-down"
              style={Object {}}
            />,
            "className": "timeline-nav timeline-nav--next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "color": "",
              "name": "chevron-down",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
